### PR TITLE
API Endpoints

### PR DESCRIPTION
**Pull request by Danwhy**
Adds endpoints for
* Adding assessments (#2)
* Listing all assessments (#5)

### CODE REVIEW

**Review by sohilpandya — CHANGES_REQUESTED** [review]
Hey @Danwhy, happy with the upgrades here, I would like us to add a couple of unit tests for ddb and a couple of integration ones for the endpoint.

File: src/AddAssessment.js [hunk 1]
```diff
@@ -22,9 +22,27 @@ class AddAssessment extends Component {
     this.setState(newState);
   }
 
+  addAssessment () {
+    let init = {
+      method: 'POST',
+      body: JSON.stringify(Object.assign({}, this.state, {user_id: "TEST"})) // TODO: replace with real user id
+    };
+
+    fetch(`${process.env.REACT_APP_API_URL}/assessment`, init)
```
Comment (sohilpandya) [inline src/AddAssessment.js]: Just a comment around extracting this logic into it's own file so that we can keep this file size down :)

Comment (Danwhy) [reply src/AddAssessment.js]: :+1: Sounds good. Do you mean abstracting the fetch logic in general, or just extracting this specific addAssessment function?

File: server/handlers/assessment.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+const AWS = require('aws-sdk');
+var ep = new AWS.Endpoint(process.env.DYNAMO_ENDPOINT);
+const dynamoDB = new AWS.DynamoDB({region: 'eu-west-2', endpoint: ep});
+
+async function add(request, h) {
```
Comment (sohilpandya) [inline server/handlers/assessment.js:5]: What is the `h` doing here? might want to be a little more descriptive with the name?

Comment (Danwhy) [reply server/handlers/assessment.js:5]: It's the new hapi [response toolkit](https://github.com/hapijs/hapi/blob/master/API.md#response-toolkit) they added in v17. `h` is what they call it, so I just went with that

File: src/AddAssessment.js (hunk 1 above)
Comment (sohilpandya) [reply src/AddAssessment.js]: the fetch logic. :)

File: server/handlers/assessment.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+const AWS = require('aws-sdk');
+var ep = new AWS.Endpoint(process.env.DYNAMO_ENDPOINT);
+const dynamoDB = new AWS.DynamoDB({region: 'eu-west-2', endpoint: ep});
```
Comment (sohilpandya) [inline server/handlers/assessment.js:3]: Just putting a question out here, would it be beneficial to memoize this ddb instance? 

I know performance for this app isn't going to be affected but just throwing it out there :)

File: server/handlers/assessment.js (hunk 2 above)
Comment (sohilpandya) [reply server/handlers/assessment.js:5]: Ohh! I've not looked at hapi in a while. 

:) 

Thanks for that!

File: server/handlers/assessment.js (hunk 3 above)
Comment (sohilpandya) [reply server/handlers/assessment.js:3]: actually, forget about it. Just a note for future if we do performance related work.

**Review by sohilpandya — APPROVED** [review]
LGTM

**Pull request merged**